src/Compilers/Core/Portable/CodeAnalysisResources.resx

```diff
@@ -450,7 +450,7 @@
     <value>Reported diagnostic has an ID '{0}', which is not a valid identifier.</value>
   </data>
   <data name="InvalidDiagnosticLocationReported" xml:space="preserve">
-    <value>Reported diagnostic '{0}' has a source location in file '{1}', which is not part of the compilation being analyzed.</value>
+    <value>Reported diagnostic '{0}' has a source location {1} in file '{2}', which is not part of the compilation being analyzed.</value>
```

Comment:
Yes, consider adding a new message string or changing the text to make it explicit that the location is invalid. Probably better to just have a separate string (you can re-use diagnostic ID).